server/src/main/java/org/jboss/as/server/deployment/module/ModuleSpecProcessor.java

```diff
@@ -146,9 +148,9 @@ private void deployModuleSpec(final DeploymentPhaseContext phaseContext) throws
      * @param module              The additional module
      */
     private void addAllDependenciesAndPermissions(final ModuleSpecification moduleSpecification, final AdditionalModuleSpecification module) {
-        module.addSystemDependencies(moduleSpecification.getSystemDependencies());
-        module.addLocalDependencies(moduleSpecification.getLocalDependencies());
-        for(ModuleDependency dep : moduleSpecification.getUserDependencies()) {
+        module.addSystemDependencies(moduleSpecification.getSystemDependenciesSet());
```

Comment:
I didn't mention this before but I really like the reduction in memory use by using Set with these changes.